Decisions.TruBot/TruBotSettings.cs

```diff
@@ -0,0 +1,214 @@
+[Writable]
+public class TruBotSettings : AbstractModuleSettings, IInitializable, INotifyPropertyChanged, IValidationSource
+{
+    internal const string TRUBOT_IMAGES_PATH = "../wwwroot/Content/CustomModuleImages/Decisions.TruBot/|trubot.svg";
+    
+    public TruBotSettings()
+    {
+        this.EntityName = "TruBot Settings";
+    }
+    
+    [ORMField]
+    private string baseUrl = "http://localhost:56498";
+     
+    [ORMField]
+    private string username = "";
+     
+    [ORMField]
+    private string password = "";
+     
+    [ORMField]
+    [PropertyHidden]
+    private string token;
+    
+    [ORMField]
+    [PropertyHidden]
+    private string sid;
+
+    [PropertyClassification(0, " ", "TruBot Settings")]
+    [ReadonlyEditor]
+    [ExcludeInDescription]
+    public string ApiKeyMessage
+    {
+        get
+        {
+            return "A license for TruBot is needed. Learn more at:https://www.datamatics.com/lp/intelligent-automation/idp-trucap/partner/decisions?utm_source=decisions.com&utm_medium=module_link_click&utm_content=ad1";
+        }
+        set { }
+    }
+
+    [PropertyClassification(1, "Base URL", "TruBot Settings")]
+    [DataMember]
+    [WritableValue]
+    public string BaseUrl
+    {
+        get => baseUrl;
+        set
+        {
+            baseUrl = value;
+            OnPropertyChanged(nameof(BaseUrl));
+        }
+    }
+     
+    [PropertyClassification(2, "Username", "TruBot Settings")]
+    [DataMember]
+    [WritableValue]
+    public string Username
+    {
+        get => username;
+        set
+        {
+            username = value;
+            OnPropertyChanged(nameof(Username));
+        }
+    }
+     
+    [PasswordText]
+    [PropertyClassification(3, "Password", "TruBot Settings")]
+    [DataMember]
+    [WritableValue]
+    public string Password
+    {
+        get => password;
+        set
+        {
+            password = value;
+            OnPropertyChanged(nameof(Password));
+        }
+    }
+     
+    [PropertyHidden]
+    [DataMember]
+    [WritableValue]
+    public string Token
+    {
+        get => token;
+        set
+        {
+            token = value;
+        }
+    }
+    
+    [PropertyHidden]
+    [DataMember]
+    [WritableValue]
+    public string Sid
+    {
+        get => sid;
+        set
+        {
+            sid = value;
+        }
+    }
+
+    public string GetBaseUrl(string? overrideBaseUrl)
+    {
+        string url = baseUrl;
+        if (!string.IsNullOrEmpty(overrideBaseUrl)) 
+            url = overrideBaseUrl;
+
+        return url;
+    }
+
+    public string GetApiUrl(string? overrideBaseUrl)
+    {
+        return $"{GetBaseUrl(overrideBaseUrl)}/CockpitPublicWebApi/api";
+    }
+
+    public string GetAccountUrl(string? overrideBaseUrl) 
+    {
+        return $"{GetApiUrl(overrideBaseUrl)}/Account";
+    }
+    
+    public string GetBotUrl(string? overrideBaseUrl) 
+    {
+        return $"{GetApiUrl(overrideBaseUrl)}/Bot";
```

Comment:
Are you using this method in the steps?
I commented before about consistency in building the url, these seem like they could be used in a few places.

Reply:
Yes, this is being used for a few bot steps